File: app/src/main/java/com/theappsolutions/boilerplate/util/other_utils/StreamsUtils.java
package com.theappsolutions.boilerplate.util.other_utils;

import java.io.BufferedReader;
import java.io.IOException;
import java.io.InputStream;
import java.io.InputStreamReader;

/**
 * @author Dmytro Yakovlev d.yakovlev@theappsolutions.com
 * @copyright (c) 2017 TheAppSolutions. (https://theappsolutions.com)
 */
public class StreamsUtils {

    public static String convertInputStreamToString(InputStream inputStream) throws IOException {
        BufferedReader r = new BufferedReader(new InputStreamReader(inputStream));
        StringBuilder total = new StringBuilder();
        String line;
        while ((line = r.readLine()) != null) {
            total.append(line).append('\n');
        }
        return total.toString();
    }

}

File: app/src/main/java/com/theappsolutions/boilerplate/util/other/StreamsUtils.kt
package com.theappsolutions.boilerplate.util.other

import java.io.BufferedReader
import java.io.IOException
import java.io.InputStream
import java.io.InputStreamReader

/**
 * @author Dmytro Yakovlev d.yakovlev@theappsolutions.com
 * @copyright (c) 2017 TheAppSolutions. (https://theappsolutions.com)
 */

@Throws(IOException::class)
fun convertInputStreamToString(inputStream: InputStream): String {
    val r = BufferedReader(InputStreamReader(inputStream))
    val total = StringBuilder()
    var line: String
    while (true) {
        line = r.readLine() ?: break
        total.append(line).append('\n')
    }
    return total.toString()
}

